### PR TITLE
Convert IdentityService to suspend functions

### PR DESCRIPTION
**Pull request by Dominaezzz**
Signed-off-by: Dominic Fischer <dominicfischer7@gmail.com>

### Pull Request Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [x] Pull request is based on the develop branch
- [ ] Pull request updates [CHANGES.md](https://github.com/vector-im/element-android/blob/develop/CHANGES.md)
- [x] Pull request includes screenshots or videos if containing UI changes
- [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)

### CODE REVIEW

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/identity/DefaultIdentityService.kt [hunk 1]
```diff
@@ -136,67 +130,64 @@ internal class DefaultIdentityService @Inject constructor(
...
-    override fun startBindThreePid(threePid: ThreePid, callback: MatrixCallback<Unit>): Cancelable {
+    override suspend fun startBindThreePid(threePid: ThreePid) {
         if (homeServerCapabilitiesService.getHomeServerCapabilities().lastVersionIdentityServerSupported.not()) {
-            callback.onFailure(IdentityServiceError.OutdatedHomeServer)
-            return NoOpCancellable
+            throw IdentityServiceError.OutdatedHomeServer
         }
 
-        return taskExecutor.executorScope.launchToCallback(coroutineDispatchers.main, callback) {
+        withContext(coroutineDispatchers.main) {
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/identity/DefaultIdentityService.kt:138]: I'm wondering if this is necessary to keep the `withContext()` call here (and for other methods too). WDYT?

**Review by bmarty — APPROVED** [review]
LGTM, thanks for the PR.
Just one question about the usage of `withContext` (but maybe out of scope of this PR)
Benoit

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/identity/DefaultIdentityService.kt (hunk 1 above)
Comment (Dominaezzz) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/identity/DefaultIdentityService.kt:138]: Yeah I was thinking about this. We can (and should) drop the `withContext` here.
To keep things simple I left it in, to mostly preserve previous behaviour.
Once all the callbacks have been converted and everything is consistently marked as `suspend`, I was planning to review all the uses of `withContext` and remove the unnecessary ones.
Although....... most of these have been pretty simple so far but I only realized that after like the 9th PR, by which point it was too late 😂.
We can just remove them; I'm at work right now so I can do so in the evening.

Comment (Dominaezzz) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/identity/DefaultIdentityService.kt]: Removed

**Review by bmarty — APPROVED** [review]
LGTM, thanks for the update. But please next time add new commits, because I had to review the whole code again since you did a squash and force push including the previous changes.

**Comment by Dominaezzz** [comment]
Will do that next time. :+1:
(Sorry about the other PRs too :sweat_smile:)

**Pull request merged**